sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/functions/ScalarFunction.java

```diff
@@ -149,7 +149,7 @@ public interface ScalarFunction<R> extends BoundFunction {
    * @return a result value
    */
   default R produceResult(InternalRow input) {
-    throw new SparkUnsupportedOperationException("_LEGACY_ERROR_TEMP_3146");
+    throw new SparkUnsupportedOperationException("SCALAR_FUNCTION_NOT_COMPATIBLE");
```

Comment:
The error condition requires the parameter `scalarFunc`. Could you pass it here, please.

Reply:
Fixed with 

```
throw new SparkUnsupportedOperationException(
            "SCALAR_FUNCTION_NOT_COMPATIBLE",
            Map.of("scalarFunc", QuotingUtils.quoteIdentifier(name()))
    );
```

I used `QuotingUtils.quoteIdentifier` instead `toSQLId` because this method is unavaible from this location.